spec/queries/keywords_query_spec.rb

```diff
@@ -30,12 +30,66 @@
       end
 
       context 'given filter_params is nil' do
-        it 'returns all keywords' do
-          user = Fabricate(:user)
-          Fabricate.times(2, :keyword, user: user)
+        it 'returns Keyword scope' do
           keywords = described_class.new(Keyword, nil).call
 
-          expect(keywords.count).to eq 2
+          expect(keywords).to eq Keyword
+        end
+      end
+    end
+
+    context 'given the user search for result_urls' do
+      context 'given the user search for a word vpn and match at least 1' do
+        it 'return a keyword' do
+          user = Fabricate(:user)
+          result_urls = ['https://www.thetopvpn.com', 'https://www.nordvpn.com', 'https://www.vnexpress.net']
+          keyword = Fabricate(:keyword, result_urls: result_urls, user: user)
+          filter_params = { word: 'vpn', match_at_least: '1' }
+          keywords = described_class.new(Keyword, filter_params).call
+
+          expect(keywords).to eq [keyword]
+        end
+      end
+
+      context 'given the user search for vpn word e and match at least 2' do
+        it 'returns a keyword' do
+          user = Fabricate(:user)
+          result_urls = ['https://www.topvpn.com/vpn', 'https://www.nordvpn.com', 'https://www.vnexpress.net']
+          keyword = Fabricate(:keyword, result_urls: result_urls, user: user)
+          filter_params = { word: 'vpn', match_at_least: '2' }
+          keywords = described_class.new(Keyword, filter_params).call
+
+          expect(keywords).to eq [keyword]
+        end
+      end
+
+      context 'given there is no matched result' do
+        it 'returns empty' do
+          user = Fabricate(:user)
+          result_urls = ['https://www.topvpn.com', 'https://www.nordvpn.com', 'https://www.vnexpress.net']
+          Fabricate(:keyword, result_urls: result_urls, user: user)
+          filter_params = { word: 'vpn', match_at_least: '2' }
+          keywords = described_class.new(Keyword, filter_params).call
+
+          expect(keywords).to be_empty
+        end
+      end
+
+      context 'given user provides word parameter and leaves match_at_least parameter nil' do
+        it 'returns Keyword scope' do
+          filter_params = { word: 'vpn' }
+          keywords = described_class.new(Keyword, filter_params).call
+
+          expect(keywords).to eq Keyword
+        end
+      end
+
+      context 'given user provides match_at_least parameter and leaves word parameter nil' do
+        it 'returns Keyword scope' do
+          filter_params = { match_at_least: 'vpn' }
+          keywords = described_class.new(Keyword, filter_params).call
+
+          expect(keywords).to eq Keyword
```

Comment:
While this works, it's quite implicit. Should we make the `.all` scope explicit?

```suggestion
        it 'returns all the keywords' do
          filter_params = { match_at_least: 'vpn' }
          keywords = described_class.new(Keyword, filter_params).call

          expect(keywords).to eq Keyword.all
```